### PR TITLE
Bump versions and migrate to AndroidX

### PR DESCRIPTION
**Pull request by narenkmanoharan**
Bumping all relevant libraries and migrate to AndroidX. This PR also adds source and target compatibility for 1.8 so this might make it so that a few consumers of the library might not be able to user. But since API 30 there is a new warning about compatibility with java 1.7 and lower so just bumping it up. Let me know anyone feels strongly against it.

Also updated the travis CI job as well with the updated dependencies.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ android:
   components:
     - tools
     - platform-tools
-    - build-tools-27.0.3
-    - android-27
+    - build-tools-29.0.2
+    - android-30
```
Comment (ryanmoelter) [inline .travis.yml:8]: `android-30` with `build-tools-29.*`? Are there `build-tools-30.*` or no?

Comment (ryanmoelter) [reply .travis.yml:8]: Also, should we use `BUILD_TOOLS_VERSION` here?

Comment (ryanmoelter) [reply .travis.yml:8]: Lol ignore me, this is in the travis file

Comment (narenkmanoharan) [reply .travis.yml:8]: build-tools-30 is in RC so i did not wanna use that yet. But can upgrade to that later. :)

**Pull request merged**